psycopg/pqpath.c

```diff
@@ -360,8 +360,7 @@ pq_begin_locked(connectionObject *conn, PyThreadState **tstate)
     }
     else {
         snprintf(buf, bufsize,
-            conn->server_version >= 80000 ?
-                "BEGIN%s%s%s%s" : "BEGIN;SET TRANSACTION%s%s%s%s",
+                 "BEGIN%s%s%s%s",
```

Comment:
I had this bit in mind more than any other.

Every other change is just about skipping tests, and about raising a cleaner error rather than an internal one because some function/construct doesn't exist.

But this is a different guy: if we drop this, psycopg will just stop working on these old servers (unless in autocommit).

I think I would leave the test here, because if someone is called to do some archeological work on some half-forgotten server deep in the heart of a temple in a jungle, then they could still use psycopg to operate on it, otherwise they'll also need to dig out an old psycopg version, or to do something else dangerous, which might result in a huge boulder rolling over them.